### DISCUSSION TITLE
Adding <NodeResizer/> to custom nodes.

### DISCUSSION THREAD

**Discussion by AmmarZeer** [xyflow/xyflow#2735] [Q&A] [answered]
I've implemented resize functionality on the nodes using Movable.
I want to remove the code I wrote for Movable and replace it with <NodeResizer/> component.
` <NodeResizer nodeId={data.id} isVisible={selected} />`
I've added this line inside my custom node, but whenever I click on the node, I get a white screen (crash) and the error says:
Uncaught Error: [React Flow]: Seems like you have not used zustand provider as an ancestor.
I have had this error before and it was caused by not wrapping the function with <ReactFlowProvider>, but I already have it wrapped with ReactFlowProvider, what causes this error for <NodeResizer/>

**Comment by bcakmakoglu** [comment, marked as answer]
Are you using the same versions for the core package and node resizer?
This issue can appear if you're not using the same version for both of them.

**Reply by AmmarZeer** [reply]
Yes I've migrated to V11 of reactflow

**Reply by bcakmakoglu** [reply]
Yeah but they have to be the same versions, i.e. if you use [11.4.0](https://github.com/wbkd/react-flow/releases/tag/11.4.0) you need `11.4.0` for the NodeResizer as well :)

Are you sure your versions match up?

**Reply by AmmarZeer** [reply]
I was on reactflow v11.3.2 when I first used the component, then I've noticed the docs saying "you need reactflow >= v11.3.3 for this to work".
I updated the package but am still receiving the same error.

**Reply by bcakmakoglu** [reply]
Mh... my bad it seems the version numbers aren't using matching pattern as I assumed they did, I guess that only applies to the controls, background etc. packages (though there seem to be missing versions on npm for those as well)
@moklick Did I misunderstand the version patterns? I thought they had to match up for Zustand/ReactFlowProvider to work properly?

@AmmarZeer If you're using `11.3.3` you need the NodeResizer pkg at `1.1.0` not `1.2.0` (the latest version)
If you update to `11.4.0` you need to install the NodeResizer at `1.2.0`

**Reply by AmmarZeer** [reply]
@bcakmakoglu yes turns out i need to npm install the noderesizer again after i updated reactflow package to 11.4.0

1 more thing though, in the example that the library provides, the div's height and width always adjust to the NodeResizer component, when I do it though the div inside the custom node dimensions stay the same, is there something am missing on how the NodeResizer works?

**Reply by bcakmakoglu** [reply]
I'm not sure if I follow correctly.
The `.react-flow__node` element will have `width` and `height` applied.
If you have a wrapping div inside your custom node, it won't have those values applied as well (as it's applied to it's parent, the `.react-flow__node` element).
Or am I misunderstanding you here?

**Reply by moklick** [reply]
Yep, this is a bit annoying. We should move the store stuff out of the core so you don't always need to update all packages when you update one.. 

@AmmarZeer please provide a condesandbox so we can have a look.

**Reply by AmmarZeer** [reply]
@moklick @bcakmakoglu  
I am not sure how to run a code on a sandbox so apologies but here's the custom node that implements the <NodeResizer />
```
import { memo } from "react";
import { Handle, Position } from "reactflow";
import NodeToolBar from "../NodeToolBar/NodeToolBar";
import { NodeResizer } from "@reactflow/node-resizer";
import "@reactflow/node-resizer/dist/style.css";
import styles from "./CustomNodes.module.scss";

const hideHandleStyle = {
  backgroundColor: "transparent",
  borderColor: "transparent",
};
function GroupNode({ data, selected }) {
  return (
    <>
      <NodeResizer nodeId={data.id} isVisible={selected} />
      {!data?.hidden && (
        <NodeToolBar
          nodeId={data.id}
          position={Position.Right}
          handleNodeToolBarCopy={data.handleNodeToolBarCopy}
          handleNodeToolBarDelete={data.handleNodeToolBarDelete}
        />
      )}
      <div
        className={`${styles.groupNode} resizable`}
        //styels prop is used for resizing
        style={{ ...data?.styles }}
        tabIndex={0}
        id={data?.id}
      >
        <Handle
          type="source"
          position={Position.Top}
          id={"top"}
          style={data.hidden ? { ...hideHandleStyle } : {}}
          className="hiddenInScreenShot"
        />{" "}
        {/*using the attribute hidden={true} instead of hiding the handle with style will break the canvas (edges will go wild)*/}
        <Handle
          type="source"
          position={Position.Right}
          id={"right"}
          style={data.hidden ? { ...hideHandleStyle } : {}}
          className="hiddenInScreenShot"
        />
        <Handle
          type="source"
          position={Position.Left}
          id={"left"}
          style={data.hidden ? { ...hideHandleStyle } : {}}
          className="hiddenInScreenShot"
        />
        <Handle
          type="source"
          position={Position.Bottom}
          id={"bottom"}
          style={data.hidden ? { ...hideHandleStyle } : {}}
          className="hiddenInScreenShot"
        />
      </div>
    </>
  );
}

export default memo(GroupNode);
```
and this is the result: as you can see the "node" component gets resized (since the handles position move) but the div inside it doesn't
<img width="720" alt="image" src="https://user-images.githubusercontent.com/93482734/211367766-33c5411b-b53a-4dbe-bdc8-295dc113d2bb.png">

I used to use Movable where `resizable.on(resizeEnd)` I would send the new height and width styles  through the data prop and change the dimensions of the the div inside the node like so:
<img width="640" alt="image" src="https://user-images.githubusercontent.com/93482734/211368572-d1fc42d1-9254-4b72-95fe-1e17d881f87b.png">
 
 Hope that makes sense, if it doesn't pls let me know to provide more details.

**Reply by moklick** [reply]
You don't need to apply `data.styles` to your div. We are already doing that for you on the div that your custom node is wrapped with. This shouldn't be necessary to do: `style={{ ...data?.styles }}`. `height: '100%'` should do it here.

**Reply by irfanhathoti** [reply]
i also got the same error but i can't find correct solution could you please help me this how to resolve ? and i updated all packages but its not work

**Reply by AmmarZeer** [reply]
@irfanhathoti 
Make sure to update reactflow package before installing the resizer package.
if you had the resizer installed before updating reacflow package, update the resizer package one more time and it should work as expected.

**Reply by irfanhathoti** [reply]
ya thanks , i apologies iam using old version of react-flow-render and in reactflow docs i seen in that the type of node data  storing that resizable node component but we are using type as styling of node you can check this link https://reactflow.dev/docs/api/nodes/node-resizer/#noderesizer--component could you please help me to reach out
